packages/common/src/discussions/utils.ts

```diff
@@ -1,16 +1,36 @@
 import { IGroup, IItem } from "@esri/arcgis-rest-types";
 import { IHubContent, IHubItemEntity } from "../core";
 import { CANNOT_DISCUSS } from "./constants";
+import { IRequestOptions } from "@esri/arcgis-rest-request";
+import { updateItem, updateGroup } from "@esri/arcgis-rest-portal";
+import { IUserRequestOptions } from "@esri/arcgis-rest-auth";
 
 /**
  * Utility to determine if a given IGroup, IItem, IHubContent, or IHubItemEntity
  * is discussable.
- * @export
- * @param {IGroup|IItem|IHubContent|IHubItemEntity} The subject to evaluate
+ * @param {IGroup|IItem|IHubContent|IHubItemEntity} subject
  * @return {boolean}
  */
 export function isDiscussable(
-  subject: IGroup | IItem | IHubContent | IHubItemEntity
+  subject: Partial<IGroup | IItem | IHubContent | IHubItemEntity>
 ) {
   return !(subject.typeKeywords ?? []).includes(CANNOT_DISCUSS);
 }
+
+/**
+ * Adds or removes CANNOT_DISCUSS type keyword and returns the updated list
+ * @param {IGroup|IHubContent|IHubItemEntity} subject
+ * @param {boolean} discussable
+ * @returns {string[]} updated list of type keywords
+ */
+export function setDiscussableKeyword(
+  typeKeywords: string[],
+  discussable: boolean
+): string[] {
+  const updatedTypeKeywords = discussable
+    ? typeKeywords.filter(
+        (typeKeyword: string) => typeKeyword !== CANNOT_DISCUSS
+      )
+    : [...typeKeywords, CANNOT_DISCUSS];
```

Comment:
On the off chance that things get out of sync, the second part of this will always add `CANNOT_DISCUSS` even if it's already present. I'm not sure of the risk of this occurring, but it's a possible issue. In other places I think we always remove the keyword (if present), and only add it back in if it's needed. Your call if you want to add this guard

Reply:
Agreed, would be nice to avoid duplicate entries added to the array. Additionally complicating things is that `typeKeywords` is _optional_ as defined on [IItem interface](https://github.com/Esri/arcgis-rest-js/blob/97322cabd6ad07bf0b5303b9719d2c085bde3c54/packages/arcgis-rest-types/src/item.ts#L19). Maybe this should account for that and always return an Array even if the given `typeKeywords` reference is `undefined`? This method's arg definition should probably reflect the optional nature of `typeKeywords`, given that, maybe it'd make sense to reverse the order of the args since we're dealing with positional arguments and required args can't come after optional args?

Reply:
I just verified this, but `typeKeywords` is optional for the POST to create an item... but the API will inject type specific keywords, and thus always return the `typeKeywords` as an array. What's more - even if you update the item with an empty array, the API will re-add the type specific keywords... unexpected but interesting to know.

Reply:
Nice find @dbouwman. If that's the case, maybe we're OK not checking for truthiness of the given `typeKeywords` like I proposed above and disregard my request to reorder the args add the optional indicator to `typeKeywords`